File: lib/py/uapi/internal/validation/ValidateString.py
from typing import list, Union
from uapi.internal.types.UString import _STRING_NAME
from uapi.internal.validation.ValidationFailure import ValidationFailure


def validate_string(value: Union[str, object]) -> list['ValidationFailure']:
    from uapi.internal.validation.GetTypeUnexpectedValidationFailure import get_type_unexpected_validation_failure

    if isinstance(value, str):
        return []
    else:
        return get_type_unexpected_validation_failure([], value, _STRING_NAME)

File: lib/java/src/main/java/io/github/telepact/internal/validation/ValidateString.java
package io.github.telepact.internal.validation;

import static io.github.telepact.internal.types.VString._STRING_NAME;
import static io.github.telepact.internal.validation.GetTypeUnexpectedValidationFailure.getTypeUnexpectedValidationFailure;

import java.util.List;

public class ValidateString {
    public static List<ValidationFailure> validateString(Object value) {
        if (value instanceof String) {
            return List.of();
        } else {
            return getTypeUnexpectedValidationFailure(List.of(), value, _STRING_NAME);
        }
    }
}
